server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchAction.java

```diff
@@ -155,12 +163,22 @@ private static ComponentQuery buildQuery(SearchWsRequest request) {
       .build();
   }
 
-  private enum ComponentDToComponentResponseFunction implements Function<ComponentDto, WsComponents.Component> {
-    INSTANCE;
+  private class ComponentDToComponentResponseFunction implements Function<ComponentDto, WsComponents.Component> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)